torch/utils/data/datapipes/iter/routeddecoder.py

```diff
@@ -41,7 +50,7 @@ def __iter__(self) -> Iterator[Tuple[str, Any]]:
             result = self.decoder(data)
             yield (pathname, result[pathname])
 
-    def __len__(self):
-        if self.length == -1:
-            raise NotImplementedError
-        return self.length
+    def __len__(self) -> int:
+        if isinstance(self.datapipe, Sized) and len(self.datapipe) >= 0:
```

Comment:
I think the second part of this check is unnecessary: if the datapipe is a `Sized` collection shouldn't its length always be `>=0`?

Reply:
Emmm. I was thinking about the previous datapipe can have `-1` as the length. You are right, if the prior datapipe has -1 as length, it should raise `NotImplementedError`, so the second argument can be deleted.